### PR TITLE
two LS I +61 303 papers, CTA 1 SED

### PR DESCRIPTION
**Pull request by GernotMaier**

### CODE REVIEW

File: input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# datatype:
+# - {name: e_min, unit: TeV, datatype: float64}
+# - {name: e_max, unit: TeV, datatype: float64}
+# - {name: dnde, unit: TeV-1 cm-2 s-1, datatype: float32}
+# - {name: dnde_err, unit: TeV-1 cm-2 s-1, datatype: float32}
+# - {name: dnde_ul, unit: TeV-1 cm-2 s-1, datatype: float32}
+# meta: !!omap
+# - data_type: sed
+# - source_id: 14
+# - reference_id: 2016ApJ...817L...7A
```
Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: This `reference_id: 2016ApJ...817L...7A` doesn't match the folder name: `2008ApJ...679.1427A/tev-000014-sed.ecsv`

File: input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# %ECSV 0.9
+# ---
+# datatype:
+# - {name: e_min, unit: TeV, datatype: float64}
```
Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: So far, whenever we have a published differential spectrum, i.e. `dnde`, the energy `e_ref` at which the diff flux is `dnde` was given. Is this not the case here?

File: input/data/2008/2008ApJ...679.1427A/tev-000014.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Source ID from TeGeV-cat: http://www.asdc.asi.it/tgevcat/)
+# LS I +61 303
+# VER J0240+612
+source_id: 14
+# Reference ID from ADS
```
Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014.yaml]: Remove comment "Reference ID from ADS". We have this almost everywhere, no need to comment.

File: input/data/2008/2008ApJ...679.1427A/tev-000014.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+#############################################################
+# spectral fit results
+# (spectral fit for phases 0.5-0.8)
+
+spec:
+  type: pl
+  mjd: {min: 53999.4, max: 54147.1}
+  phase: {min: 0.5, max: 0.8 }
+  norm: {val: 2.89e-12, err: 0.32e-12, err_sys: 0.6e-12}
```
Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014.yaml]: Remove the `1e-12` part here when doing the data entry, i.e. change to this:
```
norm: {val: 2.89, err: 0.32, err_sys: 0.6}
```

We might change this in the future, but for now data entry has to be done this way.

The 1e-12 is noted as a scale with the unit in the spec:
https://github.com/gammapy/gamma-cat/blob/master/input/schemas/dataset_source_info.schema.yaml#L190

File: input/data/2011/2011ApJ...738....3A/tev-000014-lc.ecsv [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+# - {name: e_min, unit: TeV, datatype: float64}
+# - {name: time, unit: MJD, datatype: float64}
+# - {name: significance, datatype: float32}
+# - {name: flux, unit: ph / (cm2 s), datatype: float64}
+# - {name: flux_err, unit: ph / (cm2 s), datatype: float64}
+# - {name: flux_ul, unit: ph / (cm2 s), datatype: float64}
+# meta: !!omap
+# - data_type: lc
+# - source_id: 14
+# - source_name: LS I +61 303
```
Comment (cdeil) [inline input/data/2011/2011ApJ...738....3A/tev-000014-lc.ecsv]: We've decided to remove `source_name` keys from data input files, and to only keep `source_id`.
So please remove it here.

The reasoning (see https://github.com/gammapy/gamma-cat/issues/86) was that it's extra work to check two pieced of info that contain the same info.

While I'm doing data entry, I'm using
https://gammapy.github.io/gamma-cat/sources.html
to look up `source_id` from names.

File: input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv (hunk 2 above)
Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: OK - wasn't aware of that. I've added the e_ref column.

File: input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv (hunk 1 above)
Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: Fixed.

File: input/data/2008/2008ApJ...679.1427A/tev-000014.yaml (hunk 3 above)
Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014.yaml]: OK - I usually don't delete comment. Maybe someone is just looking at one file from one source, then any help might be appreciated.

File: input/data/2008/2008ApJ...679.1427A/tev-000014.yaml (hunk 4 above)
Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014.yaml]: OK - didn't know. I would prefer to write explicitly everywhere the units. It makes it more transparent and doesn't require any insider knowledge.

Fixed this.

File: input/data/2011/2011ApJ...738....3A/tev-000014-lc.ecsv (hunk 5 above)
Comment (GernotMaier) [inline input/data/2011/2011ApJ...738....3A/tev-000014-lc.ecsv]: OK - I've fixed this.

File: input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv (hunk 2 above)
Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: Coming back to this: there are several publication where e_ref is not given, but differential fluxes are quoted in tables between e_min and e_max. Can't we add flexibility that in this case, e_ref is calculated dynamically? My feeling is that we add here information to the catalogue, which is not part of the publication.

Would have to discuss of course what to do:
- simple mean value
- weighted mean (with weights calculated from shape of the spectrum)
- ...

Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: Well, to me whatever we do is all pretty equally bad: at some point we're inventing an `e_ref` at which to put the published `dnde`. I think if I were to do it, I would add it here in the dataset and put a caveat as a comment that this `e_ref` was made up, not published.

If you put it in the tools (e.g. Gammapy SED reader) you're just encouraging bad behaviour to publish a measurement in a bad way. (And it is rare, I haven't seen that before and we have SEDs from > 100 papers by now).

Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: It's this spectrum, no?
http://iopscience.iop.org/article/10.1086/587736/fulltext/22376.figures.html
For such a spectrum (simple shape, bins not very wide) results don't depend strongly on what you do.
Would suggest to compute `e_ref = sqrt(e_min * e_max)`, i.e. log bin center here and add that as a column, and a one-line comment in the header saying that this is what was done.

Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: Yes, it is this paper. I have a few more on which I am working on. I can do this, and add a comment. But I still think this is not the correct solution - I don't think it is a bad way to let the tools calculate this later.

(actually where e_ref should be is not straightforward, and mostly not mentioned in publications).

Comment (cdeil) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: I see that the paper doesn't give a table with the spectral point information (like most papers). So it's not like in this case `e_min` or `e_max` or `e_ref` or `flux` or `dnde` or anything was really "published". The spectral points shown in Figure 2 are `dnde` drawn at some `e_ref` values.

Irrespective of the question if you add the `e_ref` column or not (we'll merge either way): can you add a "comment" or "url" in the header where this data is coming from?
If you look at other examples you'll see that it's always either a "url" pointing to a webpage or file with the information, or a "comment" saying something about the origin of the data (in this case I presume a VERITAS-internal webpage or from corresponding author).

Comment (GernotMaier) [inline input/data/2008/2008ApJ...679.1427A/tev-000014-sed.ecsv]: The corresponding author was a certain G.Maier and the values are from his notes... I've added a comment.

**Comment by cdeil** [comment]
Are you still working on this PR or is it ready to be merged?
(in case it's ready: please always leave a comment, so that I know when to look at it)

**Comment by GernotMaier** [comment]
It is ready to be merged - I addressed all of the issues.

**Comment by cdeil** [comment]
I've done a little bit of cleanup in master. Removed "ph" from units, I decided to consistently not use those in gamma-cat, otherwise all looked good!

**Pull request merged**